vsock_test.go

```diff
@@ -44,3 +48,45 @@ func TestAddr_fileName(t *testing.T) {
 		})
 	}
 }
+
+func TestUnblockAcceptAfterClose(t *testing.T) {
+	listener, err := Listen(1024)
+	if err != nil {
+		t.Fatalf("failed to run listener: %v", err)
+	}
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+
+		t.Log("start accept")
+		_, err := listener.Accept()
+		t.Log("after accept")
+		if err != nil {
+			return
+		}
+	}()
+
+	time.Sleep(1 * time.Second)
```

Comment:
I'm always a little hesitant to add a sleep in a test to make it pass, but I also don't have any better ideas at the moment.  Maybe @aybabtme will.

Reply:
I'd make this be:

```go
select {
case <-time.After(time.Second):
    t.Fail("should have unblocked")
    return
case <-done:
    // continue
}
```
this way the test case only sleeps for 1s when there's a problem, as opposed to _always_ sleeping for 1s

Reply:
1s seems too high, maybe 100ms so it doesnt perceivably slow down the test. otherwise, idt there's a good alternative...